### DISCUSSION TITLE
Displaying data from an API Issues

### DISCUSSION THREAD

**Discussion by bxia-paul** [quasarframework/quasar#12408] [General - Components / Directives / etc] [answered]
I was able to get my data from the API, when I console log it, I am able to see the data. However, I'm not sure how to be able to display this data in an element. I've tried all methods I've seen related to Vue.js and nothing seems to work. Any assistance is greatly appreciated!

I want to display the topics from the API, in the q-select.

 `<q-select
        color="pink"
        filled
        v-model="model"
        :options="options"
        label="Select a topic: "
        style="width: 250px"
      />`

This is the setup
`setup () {`
   ` const $q = useQuasar()
    const data = ref(null)
    const topics = ref(null)`

  `function loadData () {
    api.get('https://somedummydataapi.com/topics',{
  method: 'GET',`
  
  `headers: {
          'Access-Control-Allow-Origin': '*'
        }
})`
     ` .then((response) => {
        data.value = response.data
        topics.value = data.value[0].text
        console.log(data.value[0].text)
        console.log(topics.value)
      })`
     ` .catch(() => {
        $q.notify({
          color: 'negative',
          position: 'top',
          message: 'Loading failed',
          icon: 'report_problem'
        })
      })
  }`
  `onMounted(() => {
      loadData();
    })`

   ` return {
        data, 
        loadData,
        topics,
        model: ref(null),
} `

**Comment by MilosPaunovic** [comment]
It would be best for you to create some kind of reproduction for us to play with, or at least to share the entire component so it will be easier to investigate.

**Reply by bxia-paul** [reply]
@MilosPaunovic I've uploaded the entire component in the text file. I'm new to Vue.js and quasar so thank you in advance for your help.

**Comment by metalsadman** [comment, marked as answer]
You are mixing options api and composition api, `this.topics` does not exist in `setup` context. Either move/declare that reactive data in your setup using a `ref` or move the api call in options api's `methods` function.

P.s. not a good idea to upload a file of your code, use a markdown codes instead to paste your code with formatting, see https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet

**Reply by bxia-paul** [reply]
@metalsadman Thank you so much. I was able to get the data to display in a q-card. Now to be able to display it in the q-select I need it to be an array, I believe. Do you have reading materials that I can use as a guide for making topics and array?

The data return looks like this: 
[
  {
    id: 1,
    level: 1,
    text: "Flood"
  }
]

**Reply by metalsadman** [reply]
https://quasar.dev/vue-components/select#example--custom-label-value-and-disable-props

**Reply by bxia-paul** [reply]
@metalsadman Thank you, I was just able to figure it out. My code does what it needs to now. Much appreciated
